examples/transforms/transform.rs

```diff
@@ -0,0 +1,113 @@
+use bevy::prelude::*;
+
+use std::f32::consts::PI;
+
+struct CubeState {
+    start_pos: Vec3,
+    move_speed: f32,
+    turn_speed: f32,
+}
+
+struct Center {
+    min_size: f32,
+}
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_cube.system())
+        .add_system(rotate_cube.system())
+        .add_system(scale_sphere_proportional_to_cube_travel_distance.system())
+        .run();
+}
+
+fn move_cube(mut cubes: Query<(&mut Transform, &mut CubeState)>, timer: Res<Time>) {
+    for (mut transform, cube) in cubes.iter_mut() {
+        let forward = transform.forward();
+        transform.translation += forward * cube.move_speed * timer.delta_seconds();
+    }
+}
+
+fn rotate_cube(mut cubes: Query<(&mut Transform, &mut CubeState)>, timer: Res<Time>) {
+    for (mut transform, cube) in cubes.iter_mut() {
+        let full_turn = transform
+            .looking_at(Vec3::ZERO, transform.local_y())
+            .rotation;
+        let incremental_turn_weight = cube.turn_speed * timer.delta_seconds();
+        transform.rotation = transform.rotation.lerp(full_turn, incremental_turn_weight);
+    }
+}
+
+fn scale_sphere_proportional_to_cube_travel_distance(
+    mut transformabels: QuerySet<(
+        Query<(&Transform, &CubeState)>,
+        Query<(&mut Transform, &Center)>,
+    )>,
+) {
+    let distance: f32 = transformabels
+        .q0()
+        .iter()
+        .map(|(transform, cube)| (cube.start_pos - transform.translation).length())
```

Comment:
This deserves an explanation for users that aren't familiar with functional programming.

Normally I prefer for loops in examples for that reason, but the .fold is much nicer this way.

Reply:
I would actually disagree haha. 
I say use iterators all the way as that's the most idomatic rust code :wink:
And often it's clearer imo, since the iterator has a name where as a loop does not.

Reply:
Many of our users are coming from web / Python / C though, or are new to programming entirely ;) I tend to find beginners often get distracted from the point of the example when they hit iterator methods.

Also Cart agrees with me on that style point ;)

Reply:
I guess for examples its ok.... 😝
Though for everything else iterators all the way... 🥳

Reply:
I like to use the more functional approach, if the language has proper support for it.
But I think it is a valid suggestion, for anyone starting to learn a new Engine/Framework or even Language it is easier to concentrate on the specific aspects and not get confused by piled-up concepts from other realms.